internal/provider/provider.go

```diff
@@ -182,3 +186,22 @@ func New(version string) func() tf_provider.Provider {
 		}
 	}
 }
+
+func getClientConfiguration(apikeyOrJWT string) *client.Configuration {
+	var cfgOpts []client.ConfigurationOption
+	if vanityName := os.Getenv(CCVanityName); vanityName != "" {
+		cfgOpts = append(cfgOpts, client.WithVanityName(vanityName))
+	}
+	if username := os.Getenv(CCUsername); username != "" {
+		cfgOpts = append(cfgOpts, client.WithUsername(username))
+	}
+
+	cfg := client.NewConfiguration(apikeyOrJWT, cfgOpts...)
```

Comment:
Does validation of the required parameters happen inside of here?

Reply:
Nope, this happens at the managed-service.

Reply:
Okay that seems reasonable. Have you tested the experience from the tf provider when the auth is misconfigured? Are we surfacing the error in a straightforward way?

Reply:
Yup.

1. Invalid API key
```
priteshlahoti@Priteshs-MacBook-Pro terraform-provider-cockroach % COCKROACH_API_KEY=CCDB1_blah TF_ACC=1 go test /Users/priteshlahoti/go/src/github.com/cockroachdb/terraform-provider-cockroach/internal/provider/... -run=TestAccJWTIssuerResource -v
=== RUN   TestAccJWTIssuerResource
=== PAUSE TestAccJWTIssuerResource
=== CONT  TestAccJWTIssuerResource
    jwt_issuers_test.go:148: Step 1/3 error: Error running apply: exit status 1

        Error: Error creating JWT Issuer

          with cockroach_jwt_issuer.test,
          on terraform_plugin_test.tf line 12, in resource "cockroach_jwt_issuer" "test":
          12: resource "cockroach_jwt_issuer" "test" {

        Could not create JWT Issuer: invalid API key
--- FAIL: TestAccJWTIssuerResource (0.34s)
FAIL
FAIL	github.com/cockroachdb/terraform-provider-cockroach/internal/provider	0.798s
FAIL
```

2. Expired JWT
```
priteshlahoti@Priteshs-MacBook-Pro terraform-provider-cockroach % COCKROACH_API_JWT=eyJhbGciOiJSUzI1NiIsImtpZCI6ImM4OGQ4MDlmNGRiOTQzZGY1M2RhN2FjY2ZkNDc3NjRkMDViYTM5MWYiLCJ0eXAiOiJKV1QifQ.eyJpc3MiOiJodHRwczovL2FjY291bnRzLmdvb2dsZS5jb20iLCJhenAiOiIxMjAyMDM2NjU2MjQtdm44Zzk2NGI2dDFnbDBlaDNqNGQxZzExY3JmN2phdnIuYXBwcy5nb29nbGV1c2VyY29udGVudC5jb20iLCJhdWQiOiIxMjAyMDM2NjU2MjQtdm44Zzk2NGI2dDFnbDBlaDNqNGQxZzExY3JmN2phdnIuYXBwcy5nb29nbGV1c2VyY29udGVudC5jb20iLCJzdWIiOiIxMDAxOTg1ODczNjc0MzU2MTM1MzkiLCJoZCI6ImNvY2tyb2FjaGxhYnMuY29tIiwiZW1haWwiOiJwcml0ZXNoLmxhaG90aUBjb2Nrcm9hY2hsYWJzLmNvbSIsImVtYWlsX3ZlcmlmaWVkIjp0cnVlLCJhdF9oYXNoIjoieGREaWhSSXhCZHVCVXM5aVBMMVp0USIsImlhdCI6MTczMDI5NDc3NywiZXhwIjoxNzMwMjk4Mzc3fQ.vOPfqhurnlbOuAxmcsIasDmx1Sd30khFFc-4mYuwBkqEAC6Kv5RVyOO60Yr0lCcsUKyerwOuUN5e3O3S8GLURT3feRscVXa9WEHz9S2FMyv4hbwlogsYEgYYnqF__ISawRGuvo9n9PgSSUXgbF8PqLrfVaVNdeKDUX_pj7081cLJXV2sqbHTS854DYzjCTO5C7puDcNrK8Llci-waWLo0pxkOKibu-K06Av3WiQzsW5jKJmh1_x-EbcUFBWyoMT1mCaDSoVtzYxRhDk4CL3sXCZCEDL7opaAzqXaLfLb4BhXq6bnxNv8yJjv6ISnL_p55pdtuC_aiSmNxeg9IwngWQ TF_ACC=1 go test /Users/priteshlahoti/go/src/github.com/cockroachdb/terraform-provider-cockroach/internal/provider/... -run=TestAccJWTIssuerResource -v
=== RUN   TestAccJWTIssuerResource
=== PAUSE TestAccJWTIssuerResource
=== CONT  TestAccJWTIssuerResource
    jwt_issuers_test.go:148: Step 1/3 error: Error running apply: exit status 1

        Error: Error creating JWT Issuer

          with cockroach_jwt_issuer.test,
          on terraform_plugin_test.tf line 12, in resource "cockroach_jwt_issuer" "test":
          12: resource "cockroach_jwt_issuer" "test" {

        Could not create JWT Issuer: invalid JWT
--- FAIL: TestAccJWTIssuerResource (0.40s)
FAIL
FAIL	github.com/cockroachdb/terraform-provider-cockroach/internal/provider	0.794s
FAIL
```

3. Invalid API key and Invalid JWT
```
priteshlahoti@Priteshs-MacBook-Pro terraform-provider-cockroach % COCKROACH_API_KEY=blah TF_ACC=1 go test /Users/priteshlahoti/go/src/github.com/cockroachdb/terraform-provider-cockroach/internal/provider/... -run=TestAccJWTIssuerResource -v
=== RUN   TestAccJWTIssuerResource
=== PAUSE TestAccJWTIssuerResource
=== CONT  TestAccJWTIssuerResource
    jwt_issuers_test.go:148: Step 1/3 error: Error running apply: exit status 1

        Error: Error creating JWT Issuer

          with cockroach_jwt_issuer.test,
          on terraform_plugin_test.tf line 12, in resource "cockroach_jwt_issuer" "test":
          12: resource "cockroach_jwt_issuer" "test" {

        Could not create JWT Issuer: invalid secret provided in authorization header;
        expected either an API key or a JWT
--- FAIL: TestAccJWTIssuerResource (0.33s)
FAIL
FAIL	github.com/cockroachdb/terraform-provider-cockroach/internal/provider	0.719s
FAIL
```

```
priteshlahoti@Priteshs-MacBook-Pro terraform-provider-cockroach % COCKROACH_API_JWT=blah TF_ACC=1 go test /Users/priteshlahoti/go/src/github.com/cockroachdb/terraform-provider-cockroach/internal/provider/... -run=TestAccJWTIssuerResource -v
=== RUN   TestAccJWTIssuerResource
=== PAUSE TestAccJWTIssuerResource
=== CONT  TestAccJWTIssuerResource
    jwt_issuers_test.go:148: Step 1/3 error: Error running apply: exit status 1

        Error: Error creating JWT Issuer

          with cockroach_jwt_issuer.test,
          on terraform_plugin_test.tf line 12, in resource "cockroach_jwt_issuer" "test":
          12: resource "cockroach_jwt_issuer" "test" {

        Could not create JWT Issuer: invalid secret provided in authorization header;
        expected either an API key or a JWT
--- FAIL: TestAccJWTIssuerResource (0.35s)
FAIL
FAIL	github.com/cockroachdb/terraform-provider-cockroach/internal/provider	0.732s
FAIL
```